bindings/jupyroot/python/JupyROOT/helpers/utils.py

```diff
@@ -412,9 +412,47 @@ def _canJsDisplay(self):
 
 
     def _getJsCode(self):
+        # Add extra primitives to canvas with custom colors, palette, gStyle
+        prim = self.drawableObject.GetListOfPrimitives()
+
+        style = ROOT.gStyle
+        colors = ROOT.gROOT.GetListOfColors()
+        palette = 0
+
+        # always provide gStyle object
+        if prim.FindObject(style):
+           style = 0
+        else:
+           prim.Add(style)
+
+        cnt = 0
+        for n in range(colors.GetLast()+1):
+           if colors.At(n): cnt = cnt+1
+           
+        # add all colors if there are more than 598 colors defined
+        if cnt < 599 or prim.FindObject(colors):
+           colors = 0
+        else:
+           prim.Add(colors)
+
+        if colors:
+           pal = ROOT.TColor.GetPalette()
+           palette = ROOT.TObjArray()
+           palette.SetName("CurrentColorPalette")
+           for i in range(pal.GetSize()):
+              palette.Add(colors.At(pal[i]))
+           prim.Add(palette)
+
+        ROOT.TColor.DefinedColors()
+       
         # Workaround to have ConvertToJSON work
         json = ROOT.TBufferJSON.ConvertToJSON(self.drawableObject, 3)
 
+        # Cleanup primitives after conversion
+        if style: prim.Remove(style)
+        if colors: prim.Remove(colors)
+        if palette: prim.Remove(palette)
```

Comment:
@etejedor How can I call destructor of `palette` object - it would be memory leak otherwise

Reply:
Since you defined `palette` in this function, once the function call returns there will be no more references to the Python proxy of `palette`, and Python will automatically trigger the deletion of the C++ `TObjArray`.